lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/geo/CartesianPolygon.java

```diff
@@ -90,6 +89,67 @@ public boolean contains(CartesianPoint point) {
         return MathUtils.pnpoly(vertices.size(), verticesXValues, verticesYValues, (float) point.getX(), (float) point.getY());
     }
 
+    /**
+     *
+     * @param edge1A
+     * @param edge1B
+     * @param edge2A
+     * @param edge2B
+     * @return
+     */
+    public boolean isIntersectingEdge(Vector3d edge1A, Vector3d edge1B, Vector3d edge2A, Vector3d edge2B) {
```

Comment:
Why dont you use CartesianPoint instead of Vector3d? This way you can avoid calling `toVector3d`

Reply:
Good point, it's changed.